backend/corpora/common/entities/dataset_asset.py

```diff
@@ -51,7 +51,15 @@ def get_file_size(self) -> typing.Union[int, None]:
             logger.exception(f"Failed to retrieve meta data for '{self.url}'.")
             return None
         else:
-            return response["ContentLength"]
+            return response
+
+    def get_file_size(self) -> typing.Union[int, None]:
+        """
+        Retrieves the asset content length from the S3 object.
+        :return: The content length in bytes.
+        """
+        metadata = self.get_s3_metadata()
+        return metadata["ContentLength"] if metadata else None
```

Comment:
```suggestion
        return metadata.get("ContentLength")
```

Reply:
There is a chance that `metadata == None` in which case `metadata.get("ContentLength")` would throw an error

Reply:
You're absolutely right, discard this.

Reply:
I though the point of using `.get` was to avoid throwing a missing key error...? 🤔